internal/checkers/contains.go

```diff
@@ -0,0 +1,72 @@
+package checkers
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// Contains detects situations like
+//
+//	assert.True(t, strings.Contains(a, "abc123"))
+//	assert.False(t, !strings.Contains(a, "abc123"))
+//
+//	assert.False(t, strings.Contains(a, "abc123"))
+//	assert.True(t, !strings.Contains(a, "abc123"))
+//
+// and requires
+//
+//	assert.Contains(t, a, "abc123")
+//	assert.NotContains(t, a, "abc123")
+type Contains struct{}
+
+// NewContains constructs Contains checker.
+func NewContains() Contains   { return Contains{} }
+func (Contains) Name() string { return "contains" }
+
+func (checker Contains) Check(pass *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
```

Comment:
DRY
I just simplified the code, separated the common part

\+ easy support of negation with reusing helpers

cons: unusual check of the function name (a little at the end), I didn’t duplicate it at the beginning